### PR TITLE
reduce log level for summary api with docker-only

### PR DESCRIPTION
**Pull request by andyxning**
**What this PR does / why we need it**:
cAdvisor has an option `docker-only` and when it enabled, cAdvisor will only collect docker container usage info. However, kubelet and runtime daemon are running barely on a machine, i.e., not running in a container. So, when query kubelet and runtime container usage info from summary api, an error is logged. 

IMO, this is not an error and the log level should not be error. Every time the summary api is called, the message is logged. In order to reduce this annoying message, the log verbose has been improved to 5.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Reduce log level in summary api when query system containers
```

### CODE REVIEW

**Comment by andyxning** [comment]
ping @kubernetes/sig-node-pr-reviews @kubernetes/sig-instrumentation-misc

**Comment by andyxning** [comment]
@dchen1107 Could you please help to approve this PR or help to find someone else to give this PR an approve?

**Comment by dashpole** [comment]
@andyxning the `docker-only` flag is deprecated: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/globalflags_linux.go#L58.
I dont think the kubelet can perform resource management functions with only container metrics.

**Comment by andyxning** [comment]
@dashpole What functionality does kubelet

**Comment by dashpole** [comment]
@andyxning I double checked this.  [cAdvisor will only provide information for the "/" cgroup and docker containers](https://github.com/google/cadvisor/blob/master/container/raw/factory.go#L64) when `docker_only` is set.  This means the summary API will be missing stats for all pods, for the allocatable cgroup, `/kubepods`, for the runtime cgroup and the kubelet cgroup.

**Comment by andyxning** [comment]
> I double checked this. cAdvisor will only provide information for the "/" cgroup and docker containers when docker_only is set. This means the summary API will be missing stats for all pods, for the allocatable cgroup, /kubepods, for the runtime cgroup and the kubelet cgroup.

Thanks for the check. But i do not think it is right for kubelet to not work properly when using `docker-only` as at this moment we can still query the docker container data. Maybe we should update cAvdisor to fix this.

File: pkg/kubelet/server/stats/summary.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func (sp *summaryProviderImpl) Get(updateStats bool) (*statsapi.Summary, error)
 		}
 		s, _, err := sp.provider.GetCgroupStats(cont.name, cont.forceStatsUpdate)
 		if err != nil {
-			glog.Errorf("Failed to get system container stats for %q: %v", cont.name, err)
+			glog.V(5).Errorf("Failed to get system container stats for %q: %v", cont.name, err)
```
Comment (tallclair) [inline pkg/kubelet/server/stats/summary.go]: Seems like a throttle would be more useful than turning down the verbosity. Do we have a standard library for throttle log lines?

**Review by tallclair — COMMENTED** [review]
I'm a bit confused by the motivation for this. Why is cAdvisor running with docker_only in the kubelet?

File: pkg/kubelet/server/stats/summary.go (hunk 1 above)
Comment (andyxning) [reply pkg/kubelet/server/stats/summary.go]: > I'm a bit confused by the motivation for this. Why is cAdvisor running with docker_only in the kubelet?

The reason for this is that with newer Debian versions, systemd is used and everything is in a cgroup. When used with legacy kubelet source for heapster, we will scrape more cgroup data. With `docker-only` we can dramatically reduce the data size heapster will scrape for a large cluster.

Comment (andyxning) [reply pkg/kubelet/server/stats/summary.go]: > Seems like a throttle would be more useful than turning down the verbosity. Do we have a standard library for throttle log lines?

@tallclair I am afraid glog can not meet the demands.  We should make kubelet stats api worable with setting `docker-only`.

Comment (tallclair) [reply pkg/kubelet/server/stats/summary.go]: Yeah, OK, this is fine as a stop gap. Can you file an issues for making the Kubelet stats API compatible with the docker-only setting, and reference that issue from a comment here?

**Pull request closed without merging**